### DISCUSSION TITLE
Lock Tank if it Contains a Fluid

### DISCUSSION THREAD

**Discussion by mouse0270** [P3pp3rF1y/SophisticatedBackpacks#353] [Ideas]
I'd like to request the ability to have a tank upgrade locked in place if it contains a fluid. Basically, we learned that you could carry around the tank upgrades in your backpack and just equip them as an upgrade when you needed to fill them up. Then unequip them and put back any other upgrades you may wanted.

Then when you needed the fluid again, you just drop the upgrade in grab the fluid and unequip it again when you are done. I personally think if there is a fluid in the tank you shouldn't be allowed to remove the upgrade until you empty the tank.

Just an idea, like I said, I just discussed it with my friends on my server and we kinda have it as a be good and listen type of rule, but thought I would share this here as I am not sure if this is intended behavior for the tank upgrade.

**Comment by P3pp3rF1y** [comment]
I tend to agree with this one, but there's one thing - you may end up accidentally getting experience in the tank with magnet upgrade and no way to take it out before you get experience pump. I have actually added an upgrade clear recipe for this so that players can clear that fluid from there. But I will probably make it so that when there's a small amount of fluid in the tank (let's say less than a bucket) it will still be possible to remove it and clear it using the new recipe.
